src/main/java/io/github/moulberry/notenoughupdates/listener/ItemTooltipRngListener.java

```diff
@@ -0,0 +1,306 @@
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * NotEnoughUpdates is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with NotEnoughUpdates. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.moulberry.notenoughupdates.listener;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.github.moulberry.notenoughupdates.NotEnoughUpdates;
+import io.github.moulberry.notenoughupdates.core.util.StringUtils;
+import io.github.moulberry.notenoughupdates.events.RepositoryReloadEvent;
+import io.github.moulberry.notenoughupdates.profileviewer.GuiProfileViewer;
+import io.github.moulberry.notenoughupdates.util.Calculator;
+import io.github.moulberry.notenoughupdates.util.Constants;
+import io.github.moulberry.notenoughupdates.util.Utils;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.event.entity.player.ItemTooltipEvent;
+import net.minecraftforge.fml.common.eventhandler.EventPriority;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import org.lwjgl.input.Keyboard;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class ItemTooltipRngListener {
+	private final NotEnoughUpdates neu;
+	private boolean showSlayerRngFractures = false;
+	private boolean pressedShiftLast = false;
+	private int currentSelected = 0;
+	private boolean pressedArrowLast = false;
+	private boolean repoReloadNeeded = true;
+
+	private final Map<String, Integer> dungeonData = new LinkedHashMap<>();
+	private final Map<String, LinkedHashMap<String, Integer>> slayerData = new LinkedHashMap<>();
+
+	public ItemTooltipRngListener(NotEnoughUpdates neu) {
+		this.neu = neu;
+	}
+
+	@SubscribeEvent
+	public void slayerRngChance(ItemTooltipEvent event) {
+		if (!neu.isOnSkyblock()) return;
+		if (event.toolTip == null) return;
+		if (!Utils.getOpenChestName().endsWith(" RNG Meter") && !slayerData.containsKey(Utils.getOpenChestName())) return;
+
+		List<String> newToolTip = new ArrayList<>();
+
+		boolean nextLineProgress = false;
+		for (String line : event.toolTip) {
+
+			if (line.contains("Odds:")) {
+				if (NotEnoughUpdates.INSTANCE.config.tooltipTweaks.rngMeterFractionDisplay) {
+					fractionDisplay(newToolTip, line);
+					continue;
+				}
+			}
+
+			if (nextLineProgress || line.contains("Dungeon Score:") || line.contains("Slayer XP:")) {
+				if (line.contains("/§d")) {
+					String[] split = line.split(Pattern.quote("/§d"));
+					String textNeeded = split[1].replace(",", "");
```

Comment:
Again, hypixel messing with the formatting a lot makes me uncomfortable to have uncaught `split[1]`s laying around.